### DISCUSSION TITLE
Feature/Bug: Cannot install in subdirectory

### DISCUSSION THREAD

**Discussion by Higgs1** [requarks/wiki#3501] [Error / Bug Report]
Hello everyone!

Yes, I've read the feature request template notifying me to post on https://wiki.js.org/feedback

Yes, this is a duplicate / has already been covered:
- https://requarks.canny.io/wiki/p/support-hosting-wikijs-at-subpath
- https://github.com/Requarks/wiki-v1/issues/228
- https://twitter.com/requarks/status/1263108158320447490
- https://docs.requarks.io/troubleshooting#errors-when-using-a-subdirectory-reverse-proxy

Also, it was promised: [wiki-v1/aeb0c0f/README.md#L98](https://github.com/Requarks/wiki-v1/blame/aeb0c0fd1a7d6dc58a74e3238c4736a984bb41e7/README.md#L98)

Not having this feature is a major impediment for adoption for me, and it would greatly improve on this product. Wikijs is not as powerful without this feature.

However, I am now hosting a WikiJS instance at a subdirectory, and things work (woo!). But I'm not really sure how to assist others in accomplishing this feat...

Hmmm, so I suppose this isn't really phrased as a question, request, or report...

It'd be really nice if your WikiJS could support sub-directory installs. There we go! In lieu of that, it'd be neat if there was some place for users and developers to discuss how to make that a reality if there isn't one already.

Thanks :D 

P.S. It appears on my setup I've also modified the default timezone https://github.com/Requarks/wiki/issues/2523

**Comment by faymek** [comment]
Hello, could you show how to get things work? thanks!

**Reply by mikhail-khoroshev** [reply]
@Higgs1 please share your setup. We also faced with this. Still working on the solution.

**Comment by bellhyve** [comment]
I set up a workaround with Varnish to solve this in an upside down manner by making the webroot something else but keep wiki.js intact otherwise. My use case is that I want a front page for a mostly-wiki site and don't want to dilute referrers with subdomains, and also want to embed other apps that support subdirectories like hedgedoc. This could be adapted for Nginx easily.

```
	} elsif (req.http.host ~ "^(www\.)?my\.domain$") {
		if (req.method == "GET" && (req.url ~ "^\/?$")) {
			set req.url = "/index/";
			return (pass);
		} elsif (req.url ~ "^\/index\/") {
			return (pass);
		} else { set req.backend_hint = mywiki; }
	}
```

1. If you reach the front page with a GET, it loads /index/.
2. If you go to /index/, it loads the default backend (something besides wiki.js).
3. Anything else, you go to wiki.js.

Some options I'm considering:

- Add additional conditionals for additional backends; just don't bump into wiki.js's (well documented) reserved names.
- Wiki.js's "external links" supports relative links, so you might be able to link back and for the between backend apps like /collab/ and make it pretty smooth.
- Set a / as a redirect to /index/, rather than directly pulling from /index/, which means you don't need to bother juggling refs on the top page for whether they're loaded from / or /index/ (probably recommended unless you're weird, like me, about keeping the top URL tidy).

To make this bulletproof, it would probably be wise to also add a referrer check in case the front page does something magical with the top page when linked with a GET internally, but so far I've tested logins and all the editors/admin panels and everything is working so far.

Of course, this will work for almost all web apps, not just wiki.js.
